### DISCUSSION TITLE
Implication of Yolov8's internal representation of input images on detection of small objects on large input images

### DISCUSSION THREAD

**Discussion by Arminius4** [ultralytics/ultralytics#2167] [Q&A]
I'm interested in how input images are internally represented in Yolov8 and there doesn't seem to be a paper yet. 

Is it the case that each input gets transformed into 640x640 by means of scaling and padding? 
Or will it be the multiple of 640x640 that is closest to the original input dimensions?

This is relevant to me, because I have very large input images (4k) with objects to be detected of a dimension of less than 30px. 
If the input image would be scaled down to fit into 640, they would shrink to 4px. With that little information, yolov8 should have difficulties detecting them and I would need to deal with it, e.g. by tiling the input.

**Comment by ExtReMLapin** [comment]
legitimate question

**Reply by pderrenger** [reply]
@ExtReMLapin yOLOv8, like its predecessors, is designed to handle a variety of input image sizes. The internal representation of input images is not fixed to 640x640; instead, the network can process images at different resolutions, which is specified by the `imgsz` parameter during training and inference. For large images with small objects, such as your 4k images with 30px objects, it's indeed important to consider the implications of downscaling.

If you use the default `imgsz` of 640, large images will be downscaled significantly, which could make small objects difficult to detect. To address this, you have a few options:

1. **Increase `imgsz`:** You can increase the input image size (`imgsz`) to a larger value that is a multiple of the model's stride, which is typically 32 or 64, to reduce the downscaling effect. This will allow the network to retain more detail, which is beneficial for detecting small objects.

2. **Tiling:** As you mentioned, tiling the input image into smaller segments and processing each tile individually can help preserve the resolution of small objects. This approach can be combined with an increased `imgsz` for each tile to optimize detection performance.

3. **Use a custom dataset:** If you train YOLOv8 on a custom dataset that includes representative examples of the small objects at the desired resolution, the model can learn to detect these objects more effectively.

4. **Adjust anchor boxes:** YOLOv8 uses anchor boxes that are optimized for the dataset it was trained on. If your objects are significantly smaller than the objects in the training dataset, you may need to adjust the anchor box sizes to better match your small objects.

Remember to validate the model's performance on a representative subset of your data to ensure that it can detect the small objects effectively at the chosen resolution. If you need further guidance on configuring YOLOv8 for your specific use case, the documentation provides detailed instructions on various modes and settings.
